### ISSUE TITLE
Save As Userscript doesn't work for macros

### ISSUE THREAD

**Issue by tuckdesign** [Komodo/KomodoEdit#1010]
Go to Tools->Macros->Start recording.
Record Macro and click on 'Stop recording'
Now go to Tools->Macros->Save As Userscript

Popup to enter macro name shows up but komodo gets stucked there. Eventually I am able to enter few letters (but very slowly). At the end need to force shutdown Komodo and can not save macro.

Using: Komodo IDE, version 9.3.2, build 88191, platform macosx. Built on Thu Dec 10

**Comment by mitchell-as** [comment]
Hi, I'm not able to reproduce this on Mac OSX. When this happens, would you please post the contents of your error log? http://docs.komodoide.com/Troubleshooting/Where-does-Komodo-log-errors

Note that Komodo clears the contents of that log file every time you start it back up. So please send us the contents right after you have to force quit Komodo and before you open it again. Thanks!

**Comment by tuckdesign** [comment]
[pystderr.log.zip](https://github.com/Komodo/KomodoEdit/files/96073/pystderr.log.zip)

**Comment by Naatan** [comment]
That appears to be a clean log. Did you follow @mitchell-as's instructions? Ie. reproduce the issue, then copy the error log "before" restarting Komodo.

**Comment by tuckdesign** [comment]
Yes. I took log after 30 min Komodo was freezed and couldn't put it back in foreground on desktop.

**Comment by mitchell-as** [comment]
@Naatan is it possible for the "Save As Userscript..." dialog to start outside the KOMODO_USERDIR/tools directory? If so, perhaps the current directory (whatever it is) is a remote one that isn't accessible (hence the hang). Just an idea.

**Comment by Naatan** [comment]
Could be. @tuckdesign does the same issue happen when you hit File > Open > File?

**Comment by tuckdesign** [comment]
No, never happened there.
